keyboards/melgeek/z70ultra/rev1/rev1.c

```diff
@@ -0,0 +1,170 @@
+
+    {0, CS37_SW1, CS39_SW1, CS38_SW1}, /* RGB19 */
+    {0, CS37_SW2, CS39_SW2, CS38_SW2}, /* RGB20 */
+    {0, CS37_SW3, CS39_SW3, CS38_SW3}, /* RGB21 */
+    {0, CS37_SW4, CS39_SW4, CS38_SW4}, /* RGB22 */
+    {0, CS37_SW9, CS39_SW9, CS38_SW9}, /* RGB27 */
+};
+
+led_config_t g_led_config = {
+	{
+		/* C0     C1      C2      C3      C4      C5      C6      C7      C8      C9       C10      C11     C12     C13      */
+		{  0,      1,      2,      3,     4,       5,      6,      7,      8,      9,       10,      11,     12,     13,     }, /* R0 */
+		{ 15,     16,     17,     18,     19,     20,     21,     22,     23,     24,       25,      26,     27,     28,     }, /* R1 */
+		{ 30,     31,     32,     33,     34,     35,     36,     37,     38,     39,       40,      41,     NO_LED, 42,     }, /* R2 */
+		{ 44,     NO_LED, 45,     46,     47,     48,     49,     50,     51,     52,       53,      54,     55,     NO_LED, }, /* R3 */
+		{ 58,     59,     60,     NO_LED, 61,     62,     63,     NO_LED, NO_LED, 64,       65,      NO_LED, NO_LED, NO_LED, }, /* R4 */
+		{ NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, 14,     29,     56,     43,       57,      66,     67,     68,     }, /* R5 */
+    }, {
+        {0,  0},  {16,  0},  {32,  0}, {48,  0}, {64,  0}, {80,  0}, {96,  0}, {112,  0}, {128,  0}, {144,  0}, {160,  0}, {176,  0}, {192,  0}, {208,  0}, 
+		{0, 16},  {16, 16},  {32, 16}, {48, 16}, {64, 16}, {80, 16}, {96, 16}, {112, 16}, {128, 16}, {144, 16}, {160, 16}, {176, 16}, {192, 16}, {208, 16}, 
+		{0, 32},  {16, 32},  {32, 32}, {48, 32}, {64, 32}, {80, 32}, {96, 32}, {112, 32}, {128, 32}, {144, 32}, {160, 32}, {176, 32},            {192, 32}, 
+		{0, 48},             {16, 48}, {32, 48}, {48, 48}, {64, 48}, {80, 48}, {96, 48},  {112, 48}, {128, 48}, {144, 48}, {160, 48}, {176, 48},
+		{0, 64},  {16, 64},  {32, 64},           {48, 64}, {64, 64}, {80, 64},                        {96, 64}, {112, 64}, 
+		                                                             {0, 80},  {16, 80},  {32, 80},   {48, 80}, {64, 80}, {80, 80},   {96, 80}, {112, 80}, 
+	}, {
+	       4,       4,         4,        4,        4,        4,        4,        4,          4,         4,         4,          4,        4,         4,
+		   4,       4,         4,        4,        4,        4,        4,        4,          4,         4,         4,          4,        4,         4,
+		   4,       4,         4,        4,        4,        4,        4,        4,          4,         4,         4,          4,                   4,
+		   4,                  4,        4,        4,        4,        4,        4,          4,         4,         4,          4,        4,
+		   4,       4,         4,                  4,        4,        4,                               4,         4,
+		                                                               4,        4,          4,         4,         4,          4,        4,         4,
+	}
+};
+
+const is31_led g_is31_indicator_leds[6] = {
```

Comment:
Is there a specific reason that these are not included in the matrix, and handled that way? 

There is a specific indicator flag, and if you want to ensure that the indicator leds are not set, you can set the flags on init.

Reply:
The Num Lock, Caps Lock, and Scroll Lock status indicators need to be set separately

Reply:
This can be done from within the rgb matrix code.   Which is my point.  There is no reason to have it set outside of the rgb matrix configurator.   

In fact, that's why we added the "indicators" flag for the LEDs.  So that they can be controlled by the rgb matrix code, without users having to rely on hacky solutions. 

```c
void keyboard_post_init_kb(void) {
    rgb_matrix_set_flags(LED_FLAG_MODIFIER|LED_FLAG_UNDERGLOW|LED_FLAG_KEYLIGHT);
    keyboard_post_init_user();
}

bool led_update_kb(led_t led_state) {
    if (led_update_user(led_state)) {
        if (led_state.caps_lock) {
            rgb_matrix_set_color(x, 0xff, 0x00, 0x00);
            rgb_matrix_set_color(y, 0xff, 0x00, 0x00);
        } else {
            rgb_matrix_set_color(x, 0x00, 0x00, 0x00);
            rgb_matrix_set_color(y, 0x00, 0x00, 0x00);
        }

        if (led_state.num_lock) {
            rgb_matrix_set_color(x, 0x00, 0xff, 0x00);
            rgb_matrix_set_color(y, 0x00, 0xff, 0x00);
        } else {
            rgb_matrix_set_color(x, 0x00, 0x00, 0x00);
            rgb_matrix_set_color(y, 0x00, 0x00, 0x00);
        }

        if (led_state.scroll_lock) {
            rgb_matrix_set_color(x, 0x00, 0x00, 0xff);
            rgb_matrix_set_color(y, 0x00, 0x00, 0xff);
        } else {
            rgb_matrix_set_color(x, 0x00, 0x00, 0x00);
            rgb_matrix_set_color(y, 0x00, 0x00, 0x00);
        }
    }
    return true;
}
```